Remote Arming Pull Request

hai-phong-duong opened the pull request:
Remote Arming electrical project. This requires a design review from electrical leads.
We have consulted the electrical standard and relevant documents. 
We ignored errors that relate to the new Harwin Gecko connectors and errors that were deemed OKAY to be ignored by project lead. 

Contributors:
Manav Toor
Andrew Zheng
Hai Phong Duong

Commit:
added extra xtal

hai-phong-duong committed:
removed crystal

hai-phong-duong committed:
added new footprint for f1

Commit:
made the edgecut ratio 1.5 to 5

hai-phong-duong committed:
recreated pcb file, avoiding some weird issues i faced

hai-phong-duong committed:
added canhw_footprints to ra folder, since pcb editor had massive issues with locating the footprints for many of the connectors, read slack messages to properly configure footprint library with pcb editor

hai-phong-duong committed:
redesigned layout after meeting with manav on friday oct 10, discussed placements of root, power, and altimeter sections. made dimensions of board smaller. fixed courtyard overlap issues. currently, no traces. notes for next push: think about placement of pic18 and change pinouts if desired.

awzheng committed:
30 oct meeting, discussed layout, added footprints and mounting holes

awzheng committed:
pcb layout changes

awzheng committed:
6 nov fixed the osc

awzheng committed:
untangled some of the backside resistors

awzheng committed:
answered tracing concerns, plan to continue the tracing process on the right backside of the board by alternating daily.

hai-phong-duong committed:
changed some layouts, add lots of traces, check altimeter page mostly. also redid the layers. todo: add it in the final components into the pcb. also, add the rest of the traces, esp the signal traces ie. current sensing from CAN line to PIC18

awzheng committed:
feat: Fit status LEDs and route connectors

hai-phong-duong committed:
nov 17 push, finished traces, changes to GPIO pins

awzheng committed:
fixed aesthetics, moved LEDs, shortened effective board length, improved tracing

hai-phong-duong committed:
got rid of all fixable, relevant errors

hai-phong-duong committed:
nov 19 work session with andrew zheng, improved design in accordance with electrical standard

hai-phong-duong committed:
added changed schematic

ManavToor commented:
* Please provide a datasheet link for the Fuse F1
* Please change all the Gecko Connectors to an 8-pin connector on the schematic

* It looks like you still have the old gecko footprint, the new one has 8-pins
* You added dimensions and comments on the F.Cu layer, please move these to the User.Drawings layer
* For your mounting holes, please select a footprint that has a plated mounting hole
* One of the goal's of this respin is to make the board smaller, but some of these resistors are way too close to one another, they should be roughly one "ref des" apart. 
<img width="338" height="295" alt="image" src="https://github.com/user-attachments/assets/adf3db6c-8b7c-418b-974b-10dd60fb5478" />

* This trace could see up to 4A of current, it is way to thin
<img width="338" height="295" alt="image" src="https://github.com/user-attachments/assets/92b6211b-c643-4841-a4fc-c8cc93ccd89f" />

* A single via is not enough for all this current, use more
<img width="445" height="314" alt="image" src="https://github.com/user-attachments/assets/eed1c637-1984-49cb-abb2-949c94765ade" />

* The three traces can be combined into one larger trace. One of these traces is going to the LDO, it should be thicker since it is a power rail, and should not come from just one via (same reason as last point). The trace going to the INA should be routed separately from the rest.
<img width="395" height="435" alt="image" src="https://github.com/user-attachments/assets/7bf343d3-333a-48df-8436-96f066a19e88" />

* Once again, traces providing power should be a little bit thicker, there isn't a lot of current going through these guys but I would buff them up a bit
<img width="395" height="435" alt="image" src="https://github.com/user-attachments/assets/21c3bff7-5c43-4882-a2cd-2a13d29c96f3" />

* Don't daisy chain traces like this
<img width="395" height="435" alt="image" src="https://github.com/user-attachments/assets/43a4aad5-9fe1-443c-9442-f809356f071a" />

* Why is the trace so thin all of a sudden? It should be way thicker. Also, do not drop a via in the middle of the trace, have it off to the side
<img width="395" height="435" alt="image" src="https://github.com/user-attachments/assets/dde0b138-edf8-4c73-b2d5-85fa33b337d7" />

This is a good start, but there are a few more places where the traces are too thin or components are too close.

awzheng committed:
fixes made with review from manav

hai-phong-duong committed:
nov 22, more fixes in accordance with manav's feedback

hai-phong-duong committed:
addition to nov 22 push

hai-phong-duong committed:
nov 26, made fixes to PR with pranav's feedback

Pdada1 reviewed (changes requested):
<img width="256" height="322" alt="image" src="https://github.com/user-attachments/assets/ca22b454-bd5f-4c89-a197-d4299cba41f3" />

 vias too close to pad
<img width="304" height="129" alt="image" src="https://github.com/user-attachments/assets/786b40d2-116f-4fbf-962e-e72234714559" />

Via is too close to footprint pad here (is intersecting)
<img width="688" height="585" alt="image" src="https://github.com/user-attachments/assets/4361b854-b1c6-48ce-9e39-5784c95dda65" />


<img width="227" height="330" alt="image" src="https://github.com/user-attachments/assets/35a407c0-e3b3-484f-b756-aadeb03c2ec9" />

<img width="729" height="591" alt="image" src="https://github.com/user-attachments/assets/5b65ad99-1304-4ac6-b0b5-e85cdbec231b" />
You can reduce the daisy chaining and length of these traces by adding vias here

<img width="1168" height="685" alt="image" src="https://github.com/user-attachments/assets/2732aabe-4092-45e6-8a3a-1839c7d74394" />

https://waterloorocketry.slack.com/archives/C8VL7QCG0/p1717531280227019?thread_ts=1717530376.571639&cid=C8VL7QCG0
According to slackumentation 2 ematches in parallel draw 2.5A. Based on this 
the thickness of the firing traces at 0.25mm seems too thin
Confirm with @ManavToor 

<img width="880" height="582" alt="image" src="https://github.com/user-attachments/assets/723bd09d-39e6-43b5-9ec2-19371246cadb" />
Eliminate Daisychaining
<img width="900" height="586" alt="image" src="https://github.com/user-attachments/assets/a7afa208-c2d3-4793-91ec-bd1588546c65" />
<img width="882" height="521" alt="image" src="https://github.com/user-attachments/assets/ef6ffb00-69f5-4ad9-87df-2bd1c9ccbb0c" />
<img width="304" height="129" alt="image" src="https://github.com/user-attachments/assets/daddc3b5-3d5a-4b01-9a35-087c9ad3d99e" />

Should put in some vias for stiching between the two GND planes
https://resources.altium.com/p/everything-you-need-know-about-stitching-vias

hai-phong-duong committed:
nov 29, todo: ematch traces are too thin

hai-phong-duong commented:
@Pdada1 I made all the changes in the lastest push, except the ematch traces.
Last year, the traces for M1+ and M1_PYRO (etc.) were 0.7620mm (which is way thicker than 0.2500mm currently). 
I'm going to confirm with Manav if that's what we're doing this year.

hai-phong-duong committed:
second nov 29 push, fixed ematches according to last year's specs

ManavToor commented:
Pull ups don't need to be thick
<img width="932" height="499" alt="image" src="https://github.com/user-attachments/assets/af6f88ea-db3f-4bfc-9954-bdaa2851ad1f" />

You have some DRC errors that you need to fix ie
<img width="466" height="279" alt="image" src="https://github.com/user-attachments/assets/ce7f5154-a4f0-4f30-8f38-ef6c81fead7b" />
<img width="875" height="599" alt="image" src="https://github.com/user-attachments/assets/8d1f1af2-17f7-4ddb-b5e3-5625825ca1e0" />

When switching layers with thick traces, try using multiple vias (2-3)
<img width="441" height="372" alt="image" src="https://github.com/user-attachments/assets/4b95d753-a8ee-44b1-a7aa-a476d0787af8" />

The thinnest points in this fill defeat the purpose of using the copper fill in the first place. Also once again, use more vias on when switching sides (refering to the right hand side)
<img width="1365" height="348" alt="image" src="https://github.com/user-attachments/assets/1821779d-b05a-46c9-b28e-dd9b84e98254" />


Add some silkscreens: label the connectors for what they do,  add the "MCLR" indicator on the programming header, label the buzzers, and batt inputs (batt1, batt2). Also name the board,, add the waterloo rocketry logo as well as your names (refer to the previous copies of RA).

If you have time, it's more aesthetic to have rounded corners.

hai-phong-duong committed:
nov 30 push with manav's feedback

hai-phong-duong commented:
Done. I'm waiting on clarification for the trace thickness of the jumpers.

"Mochi" :P

<img width="2078" height="1020" alt="image" src="https://github.com/user-attachments/assets/b8e2222f-fafc-4efa-a818-3fa70450e7ec" />
<img width="1599" height="675" alt="image" src="https://github.com/user-attachments/assets/913a0aea-9c76-4e52-b1a9-6aa50a5f09e7" />

hai-phong-duong committed:
confirmed jumper trace thickness. improved internal track width of BATT1/2 pours. relabeled buzzers.

hai-phong-duong committed:
silkscreen fixes

Pdada1 commented:
<img width="1255" height="763" alt="image" src="https://github.com/user-attachments/assets/33ad9a8a-1709-4864-8d43-205eaeff815c" />
Reminder to update your design rules checker constraints to match the electrical standard. 
<img width="997" height="644" alt="image" src="https://github.com/user-attachments/assets/20924e97-a4d8-4f1d-8d62-7e7b7bd73b2f" />
With this added there are a lot of clearance violations and min thickness to look at

StarlightDescender reviewed (changes requested):
Sorry it took me so long to get around to this, thought I posted a review earlier but had only written it.
- 
<img width="683" height="886" alt="image" src="https://github.com/user-attachments/assets/140c4d5d-42ed-40ab-a0ab-6f484f832e39" />
still a bunch of daisy chaining on the resistor pullups for the arming FETs, it should be pretty easy to fix.
- Pad 3 of Q4 and Q5 could use a second via on that trace
- silkscreens are a bit messy, some are under components (F1, polarity indicators on the battery terminals), in general we want to have them all facing the same way but that's not super important.
- Also nitpicky but rename IN1.Cu and IN2.Cu to GND and 5V for clarity

hai-phong-duong committed:
completed ash and pranav's feedback

hai-phong-duong committed:
silkscreen fixes

Pdada1 reviewed (changes requested):
Good work to push this out, just some small changes and then I think it is pretty much good to approve

Nit
<img width="651" height="475" alt="image" src="https://github.com/user-attachments/assets/68adc372-0780-40fa-a82b-b6ecf397b322" />
Mirror this horizontally, makes it easier to follow if input to output flows from left to right

<img width="665" height="718" alt="image" src="https://github.com/user-attachments/assets/ffb9d503-9625-4b56-a4c3-1624eead50fb" />
Also run the cleanup tracks and vias tool to get rid of stray traces you made during design
<img width="1246" height="695" alt="image" src="https://github.com/user-attachments/assets/7828021b-77ec-41c8-a5e2-e5abe9aaf525" />
Change the spoke angle to 90 here to get a better gnd connection
<img width="1225" height="654" alt="image" src="https://github.com/user-attachments/assets/278a5c7d-b03c-469a-8572-41700df66c7d" />
same with here
<img width="1090" height="410" alt="image" src="https://github.com/user-attachments/assets/7d7101a9-42a9-4eba-875c-952860afd175" />
daisychaining, easy fix below
<img width="526" height="391" alt="image" src="https://github.com/user-attachments/assets/49c6969c-e377-454f-8734-567e7d87138d" />
<img width="528" height="411" alt="image" src="https://github.com/user-attachments/assets/cff94e6c-ee7d-44a8-a30e-e6191c415769" />
same idea here
<img width="506" height="507" alt="image" src="https://github.com/user-attachments/assets/2a3e6b9a-5019-4682-bda7-c7562cd95f29" />
daisychaining
<img width="1148" height="179" alt="image" src="https://github.com/user-attachments/assets/bc359f1d-e42b-40b1-9fdb-95ff6111e0a0" />
give yourself more clearance here
<img width="605" height="664" alt="image" src="https://github.com/user-attachments/assets/4dba01eb-18f7-44ed-b61c-cd7ddc78d1f2" />
daisychaining
<img width="605" height="664" alt="image" src="https://github.com/user-attachments/assets/70b483b5-aea1-4b97-9a72-5921f0124353" />
<img width="380" height="245" alt="image" src="https://github.com/user-attachments/assets/3886010f-72ef-4eda-acbc-9d7e2b831e87" />
<img width="272" height="418" alt="image" src="https://github.com/user-attachments/assets/ac83efb9-be2a-4666-8e4c-3be99baa2def" />
Give yourself a bit more clearance between via and pad on these

hai-phong-duong committed:
dec 14, pranav's feedback

Pdada1 commented:
<img width="607" height="502" alt="image" src="https://github.com/user-attachments/assets/ab7fcc9c-4ef7-48db-83d4-179cdfd9f8ed" />
maybe a rotated layout like this might work?

Pdada1 commented:
<img width="420" height="475" alt="image" src="https://github.com/user-attachments/assets/22468f7b-c1bc-46ee-b6e1-fc0e361f4f61" />

awzheng committed:
pranav feedback pt2, improved copper flow around CAN chips. todo: check drc on J5

Pdada1 reviewed (approved):
I don't see any more changes that need to be made. 
LGTM

StarlightDescender reviewed (changes requested):
Looks good overall! a few changes
- team wants to switch to 0603 for passive components (resistors capacitors), i know this is a bit annoying but placements will still the same, you'll just need to swap the footprints and then extend the traces a little bit.
- some daisychaining/90 degree traces
<img width="1016" height="346" alt="image" src="https://github.com/user-attachments/assets/8c425e28-8b4b-4a6b-a22b-e36836e76d2f" />
<img width="1042" height="755" alt="image" src="https://github.com/user-attachments/assets/a0b13873-75e8-4156-ab16-257029c43df3" />
<img width="490" height="581" alt="image" src="https://github.com/user-attachments/assets/f3ed696a-a82a-4958-bcd8-5d46120a7859" />
- Please label the pyro connectors on both sides, the mag switches need to be on the side that face out so the current connector silkscreens are facing into recelec
- also move the resistors a bit further from the plated holes for the mag switches

dare-eee-enn commented:
<img width="1111" height="649" alt="image" src="https://github.com/user-attachments/assets/5c277c86-6e27-45cc-a7e1-16a0ab0a6c4f" />
<img width="691" height="331" alt="image" src="https://github.com/user-attachments/assets/1c01aeed-c8b6-41f2-8689-d73aed9bb491" />

A bit of daisy chaining

Pdada1 committed:
Passives to 0603

Pdada1 committed:
Fix Silkscreen

Pdada1 committed:
Fixed Silkscreen Fully

Rio6 committed:
Move more passives to 0603

Lucifersan committed:
fixed footprints and silkscreen overlaps, I think. idk hopefully didnt break it

Lucifersan committed:
moved the buzzer and managed to fix a bunch of drc errors

Rio6 committed:
separate current sensing line for U7

Pdada1 commented:
<img width="1114" height="534" alt="image" src="https://github.com/user-attachments/assets/b0f474a0-cbf7-4440-8ce0-193fde1e29f1" />
There are currently a lot of DRC errors for the capacitor and resistor footprints which are marked as differing from the library footprints. Is this intentional?

Pdada1 commented:
<img width="895" height="717" alt="image" src="https://github.com/user-attachments/assets/d3145e91-0fb5-458f-b3bb-2809d19980f9" />
Why not just via directly

Pdada1 committed:
Fixed some silkscreen and minor routing changes

StarlightDescender committed:
tidy up RA folder, delete spare footprints, tidy up routing, create step and prod files

miles-milne commented:
Board looks pretty good to me. Here's just a couple of my nitpicks. 





Just some more Q3 daisy chaining
<img width="670" height="546" alt="image" src="https://github.com/user-attachments/assets/004d0e36-21a0-4578-a838-f4103e5c85bb" />



If possible, have Kelvin routing R14
<img width="865" height="851" alt="image" src="https://github.com/user-attachments/assets/77755e0e-0f5e-4dbc-bb05-f81efb04f328" />



I'd recommend giving C1 an 0805 footprint since they are often of better quality than their 0603 counterparts. 
<img width="744" height="763" alt="image" src="https://github.com/user-attachments/assets/9b8533fe-7618-4751-bb12-e0c1d7f60440" />



Silkscreen labels for J5 and J8 should not be obstructed by connectors if possible. 
Also "Alt 1 Pyro" having a different text thickness lol 
<img width="927" height="383" alt="image" src="https://github.com/user-attachments/assets/846a0c54-e72f-4f35-ba7f-5a5c40b332b4" />



The ground connections to some of the connectors (J5 & J6) are a bit dodgy due to the thermal reliefs and their small pad size. I'd recommend removing the thermal reliefs entirely for a better ground connection. 
<img width="668" height="522" alt="image" src="https://github.com/user-attachments/assets/d0aaabda-65e2-4881-9ae4-45fd2f3fc039" />
<img width="1213" height="1110" alt="image" src="https://github.com/user-attachments/assets/684972a7-32a2-4b9d-b9b5-82ffc20b11de" />

Pdada1 committed:
Fixed Miles Changes and ran via and trace cleanup, need to regen gerbers and prod files

dare-eee-enn commented:
- PIC datasheet says that R1 should be within 6mm of the MCLR pin (pg 14)
- add a datasheet or calculation for crystal
<img width="776" height="467" alt="image" src="https://github.com/user-attachments/assets/98f3c7c4-e1cb-4cb3-a57a-8edf4ce804f8" />

- raw_bat 1 and 2 traces not fat enough as per rocketry electrical standard
<img width="1352" height="115" alt="image" src="https://github.com/user-attachments/assets/60cc142b-09ee-49cd-8b3e-a7abf94a7701" />
<img width="478" height="487" alt="image" src="https://github.com/user-attachments/assets/659e1f29-4615-420b-b1e7-bf970e4b2c65" />
<img width="1002" height="633" alt="image" src="https://github.com/user-attachments/assets/3218681c-dece-4fd2-8980-c51d66a028a9" />

- you should put the waterloo rocketry logo on

StarlightDescender committed:
add logo, move R1, regen gerbers and step

dare-eee-enn committed:
Updated schematic PDF

dare-eee-enn reviewed (dismissed):
LGTM

Pdada1 commented:
<img width="823" height="562" alt="image" src="https://github.com/user-attachments/assets/3b487eeb-780b-4acb-a29c-723fd3649fb0" />
not the most ideal, but via down on one side and shifting of lower taraces to allow for the connection to the fill on the other side. 
MCLR resistor also kept within 6mm

Pdada1 committed:
Fixed unconnected gnd pour while keeping MCLR resistor wihin 6mm, also increased gnd connection to can transciever

StarlightDescender committed:
regen build files + minor aesthetic touchups

Pdada1 commented:
<img width="649" height="333" alt="image" src="https://github.com/user-attachments/assets/2f8d6291-a6d3-43f2-a8c4-b4d5f7d4f902" />
Just looking at the cad and for integration. It would probably be good to swap the buzzer position and this connector, so that this board is only constrained on two sides

Pdada1 committed:
Changed Connector Location for Better Integration

Pdada1 committed:
Fixed Board Outline Error

dare-eee-enn committed:
Changed some silkscreens

sky46 committed:
Remove second circuit from RA schematic

sky46 committed:
Initial RA layout

sky46 committed:
Smaller layout

sky46 committed:
Minor layout changes

sky46 committed:
A few RA layout changes and a tiny bit of routing

StarlightDescender reviewed (changes requested):
A few layout notes
- I would change the crystal to an SMD crystal (take a look at power board or gps board for schematic and layout examples) to save space to make routing easier
- R16 and U8 should be moved to resemble something like this
<img width="710" height="413" alt="image" src="https://github.com/user-attachments/assets/a4d94a78-90a5-45d3-8ef2-c47c3b51d7ba" />

- same with U6 and R14
- 6 pin gecko connectors (J5 and J8) too far back from board edge
- We could turn the two solder jumpers into a single jumper to save a bit of space, and there's no scenario I can imagine where we'd want to explicitly pull up one channel but leave the other channel open

sky46 committed:
More routing

sky46 committed:
Completed initial version of routing

ManavToor reviewed (commented):
Why is the GND routed like that (U1)?
<img width="492" height="564" alt="Image" src="https://github.com/user-attachments/assets/c3072f42-b1ec-4820-8214-6a85c530d94b" />

The copper connection is too thin for the amount of current going through, I would change the thermal connection to solid
<img width="526" height="620" alt="Image" src="https://github.com/user-attachments/assets/74960d27-d34e-44af-a3ff-8cba6820b783" />

Just make the power plane look like this
<img width="1305" height="502" alt="Image" src="https://github.com/user-attachments/assets/501a3817-6c68-43a6-b57a-e98d356dd3e7" />

I know it's tight but the LDO needs to be routed better, try something like this
<img width="1254" height="693" alt="Image" src="https://github.com/user-attachments/assets/e67c13b5-32a7-4dfe-84b1-3931c518a20b" />

sky46 commented:
@ManavToor I think you're on the wrong version? Should look like this:
<img width="1865" height="1087" alt="image" src="https://github.com/user-attachments/assets/6fd4041f-d6ca-458d-815e-a250b3e37cf6" />

My LDO routing is also probably a little dubious though so I'll try and touch it up:
<img width="354" height="358" alt="image" src="https://github.com/user-attachments/assets/750523fb-5ec2-4dc2-b6db-441b8800dc44" />

sky46 committed:
Make review changes

ManavToor reviewed (changes requested):
Bruh...my bad.

A good start to the board.

Please fix all the DRC warning and errors, especially the broken board outline and overlapping silkscreens.

Also, why are there silkscreens outside of the board?

You might have interference between wires to the 9V and the wires to the strat. Ask @Pdada1 if this might be a problem
<img width="600" height="504" alt="Image" src="https://github.com/user-attachments/assets/d283e611-39bd-484f-918f-4bf48b806e9e" />

In general the routing seems very sloppy, please take a little more care into how you place components and traces:
- Why is this resistor so far away from where it needs to be
<img width="537" height="373" alt="Image" src="https://github.com/user-attachments/assets/a1653523-a91c-4cdb-adc8-ba6b2cf6139b" />

- This trace could be straightened out
<img width="555" height="632" alt="Image" src="https://github.com/user-attachments/assets/794644e3-c9e6-4583-9c49-a1580999add6" />

- Try not to do this, place the via to the side of the trace
<img width="323" height="199" alt="Image" src="https://github.com/user-attachments/assets/68dd0677-5fbf-4acf-aae5-1a46e9466817" />

- This cap needs to be placed better, also it should be near pin 3 not 5
<img width="773" height="495" alt="Image" src="https://github.com/user-attachments/assets/298f5a68-3903-4992-8ce7-cf6f84be19e6" />

This is the best way to route a decoupling cap
<img width="720" height="622" alt="Image" src="https://github.com/user-attachments/assets/1bc10782-f30a-45e3-8e1f-d881c505a9fd" />

There's a few things wrong here:
- The LDO output goes from a nice thick trace to the diode to a super thing one headed into one via! Please try to make it look more like the one I showed in the previous comment (the one for the wrong version of the board)
- Why are so many things getting their 5V from that one via, defeats the purpose of a 5V plane
- The routing into that bottom resistor is sloppy
<img width="662" height="585" alt="Image" src="https://github.com/user-attachments/assets/32e7a01f-74bf-4151-bfd2-5abd05a077ea" />

Don't put a via between a diff pair if you can help it, also why is the decoupling capacitor nowhere near the power pin
<img width="751" height="573" alt="Image" src="https://github.com/user-attachments/assets/4d9fea6b-1abd-4981-81bf-4121b8f9405b" />

sky46 committed:
Manav's requested changes

sky46 committed:
A couple more minor routing changes

ManavToor reviewed (changes requested):
Looking much better!

- Placing a via in the middle of a trace normally doesn't cause any actual electrical issues but isn't best practice is all, you're placing a hole in the middle of your wire. In higher speed designs it can seriously degrade your signals.

- The decoupling capacitors are looking much better but you don't need to place a keep out zone around them
ie This can be routed as follows:
<img width="580" height="512" alt="Image" src="https://github.com/user-attachments/assets/eae7cd06-d8f7-4647-b6ea-6d84b7b6fc6d" />
<img width="609" height="467" alt="Image" src="https://github.com/user-attachments/assets/09316d41-c327-4b3d-b033-5838fd846674" />
Same thing here and a few other places, that keep out zone can be removed
<img width="587" height="465" alt="Image" src="https://github.com/user-attachments/assets/906390cc-654e-4627-a5bc-289ab6fe7b76" />

- Could you remove the squiggle
<img width="944" height="529" alt="Image" src="https://github.com/user-attachments/assets/d41dba0c-27f7-4ca8-80e4-e8be739f7874" />

- LDO routing is done well

Pdada1 commented:
<img width="903" height="604" alt="image" src="https://github.com/user-attachments/assets/1094c5cf-d748-4642-a8ad-5b3a0c9a73f9" />
From a usabilty perspective, it would be nice if the position of the battery header and the magswitch were swapped so that only two side of the board are required to be open for wires to run in. 
<img width="1117" height="412" alt="image" src="https://github.com/user-attachments/assets/d3731613-17b1-43b3-9814-93cf9077f6ed" />
can probably use pin 15 for this instead.

ManavToor commented:
You don't need to length tune crystals, not sure why it's like that on other boards. The traces should be short (and ideally symmetrical) as possible for protecting from interference. But none of this even matters at 12mhz so do whatever you'd like tbh.

sky46 committed:
Manav, Pranav requested changes.
- Switch battery and magswitch locations
- Change curr_amp_can to pin 15
and others

Pdada1 commented:
<img width="1444" height="589" alt="image" src="https://github.com/user-attachments/assets/c830abb1-8a2b-4b92-a4fa-9b02c6e0e577" />
The M1+ trace is part of the main firing circuitry, so it should be as thick as the m1 pyro. 
<img width="684" height="640" alt="image" src="https://github.com/user-attachments/assets/040597c4-e33b-4b28-b185-603d2ba976f6" />
Nit, move the via a bit and make the traces straight. 
<img width="1036" height="624" alt="image" src="https://github.com/user-attachments/assets/f602a43f-a938-4266-8837-b5973a5dd76f" />
Not a super big deal, but I think you can reduce the length of the first trace to the fuse if you use the empty space you have next to the connector. 

<img width="1051" height="172" alt="image" src="https://github.com/user-attachments/assets/94682a1d-f8e0-4691-bf67-9d5d01c29449" />
<img width="873" height="684" alt="image" src="https://github.com/user-attachments/assets/1e2ee290-f96d-458a-8edb-3471a5af5a2e" />
@StarlightDescender correct me if I'm wrong but the Strato TX requires a uart module so it should be connected to a pin in B or C port set. 

Also @Lucifersan are there signals for which having testpoints for would be useful. 

Otherwise seems good to me

Pdada1 reviewed (commented):
See above

StarlightDescender reviewed (changes requested):
- recieving from strato needs a uart peripheral so that needs to connect to a B or C pin
- put testpoints on UART_RX and the output of both op-amps (op amp outputs are nice but not strictly necessary), these can be small pads (circle or square 2mm)
- 3d model for D2 is not lining up with the pads
- 

<img width="772" height="414" alt="Image" src="https://github.com/user-attachments/assets/e5afd3bf-3352-44b7-a917-798ff7a571e6" />
ground and batt traces from the LDO don't need to be particularly large, max possible current is 500mA

more to come

sky46 committed:
Pranav and Ash requested changes.
- Thick M1/D1+ trace
- Move around CAN layout to reduce trace length
- Connect SL_TX to PORTB since it requires UART
- Add test points on SL_TX and output of current sensing op-amps for CAN and battery
- Fix alignment of D2 3d model
- Thinner ground and batt traces from LDO

sky46 commented:
Made all of Pranav's and Ash's requested changes, except finding 4 different buzzer options and adding to schematic comments (will do tomorrow). @StarlightDescender Is the new routing for ground and batt for the LDO okay?

dare-eee-enn commented:
<img width="757" height="437" alt="image" src="https://github.com/user-attachments/assets/7518ef0c-f6cf-4583-9c2a-4ef8ec0b8c00" />

- I feel like we shouldn't daisy chain these decoupling capacitors so the board doesnt completely die if the pad rips off or something
- Also capacitor is really close to the can transceiver so that might want to be moved away a little  

<img width="502" height="593" alt="image" src="https://github.com/user-attachments/assets/e928b3bb-dd59-4fe2-90fc-41ae52f2c161" />

- Same thing here for the current sense

<img width="503" height="736" alt="image" src="https://github.com/user-attachments/assets/e09248aa-0b55-4df9-8d6b-821aeba66bb1" />

- again here

<img width="1062" height="582" alt="image" src="https://github.com/user-attachments/assets/06e8733b-c35e-4699-938a-39359b64f2f9" />

- move via out of the way and move the decoupling cap close to the mclr pin

<img width="861" height="335" alt="image" src="https://github.com/user-attachments/assets/92cafbfd-3ff6-4760-8ec1-67d9ccd78280" />

- Might want to increase number of spokes or widen the spokes to ground pin of battery

jhuang2006 commented:
@StarlightDescender mounting holes shouldn't be grounded anymore right?
<img width="2305" height="1400" alt="image" src="https://github.com/user-attachments/assets/af77885e-a617-4317-bb01-0d2fcab0adc5" />

maybe we can find a mosfet with lower Rds and lower gate voltage (like 4.5V for min Rds). Though this will work just fine lol.
<img width="737" height="725" alt="image" src="https://github.com/user-attachments/assets/7f0c6393-def6-4c5a-bebf-d3e2a92ba468" />

you want a 2x 100nF caps, one for VDD and VIO in addition to the 10uF bulk (not strictly necessary). 10uF should probably also be 0805 for derating reasons.
<img width="1899" height="1025" alt="image" src="https://github.com/user-attachments/assets/5e2321e4-28bc-4487-803c-ab17f5d6d2b5" />

Why is R24 15k? can it not be 10k
<img width="1013" height="760" alt="image" src="https://github.com/user-attachments/assets/ca4965cd-f09a-4e2f-b24f-f8cba4f9d059" />

thermal reliefs look kinda small
<img width="1156" height="977" alt="image" src="https://github.com/user-attachments/assets/1d228a42-cae0-446e-98c3-15b23e1e6147" />

I may be missing something here, but it seems theres only 1 battery, so is D2 necessary? Did we historically have 2 batteries which is why the diode exists?
Also fuse the battery pls
<img width="1662" height="1262" alt="image" src="https://github.com/user-attachments/assets/e1b57eee-aa0f-4aba-9794-b45fe794ca54" />

jhuang2006 commented:
100mA enough?
<img width="1446" height="595" alt="image" src="https://github.com/user-attachments/assets/9aae536a-5d8d-4d9f-95e4-06df1475fae8" />

sky46 committed:
Darien, Ash, Jason H requested changes, silkscreen todo.
- Add 4 different frequency buzzers to schematic (same footprint)
- Don't daisy chain decoupling caps and move them a little farther from chip on LDO, current sense, CAN
- Widen battery ground spokes
- Add 100nF caps on VDD and VIO of CAN transceiver, keep 10uF and change to 0805
- Change R24 to 10k
- Remove D2 since we only have 1 battery now

sky46 committed:
Fix silkscreens and make them all the same rotation

sky46 committed:
Fix silkscreen date

sky46 committed:
Change to 500mA LDO

sky46 committed:
Add diode for LDO and add 3d model for LDO

sky46 committed:
Swap positions of geckos

sky46 committed:
Ash requested changes
- Add low-pass filter to can current sense output
- Change decoupling caps for CAN transceiver back to 0603
- Thermal reliefs in LDO filled zones
- Daisychaining

sky46 committed:
Darien requested changes.
- Switch to 250mA polyfuse on LDO 5V output
- Switch caps to 0603
- Switch R11 to 0603

StarlightDescender committed:
gen prod files

StarlightDescender committed:
final 3d model, prod files, pdf
